app/domain/authentication/authn_jwt/jwt_fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+
+    JwtFetchProviderUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        provider_uri
+        discover_provider
+        fetch_provider_keys
+      end
+
+      private
+
+      def provider_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ProviderUriResourceNameConfiguration.new(resource_id))
+        @provider_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @provider_uri ||= @provider_uri_secret[resource_id]
+      end
+
+      def discover_provider
+        discovered_provider
+      end
+
+      def discovered_provider
+        @discovered_provider ||= @discover_identity_provider.(
+          provider_uri: @provider_uri
+        )
+      end
+
+      def fetch_provider_keys
+        jwks = {
+          keys: @discovered_provider.jwks
+        }
+        algs = @discovered_provider.id_token_signing_alg_values_supported
+        @logger.debug(LogMessages::Authentication::OAuth::FetchProviderUriKeysSuccess.new)
+        OAuth::ProviderKeys.new(jwks, algs)
```

Comment:
yet again a very common code. can we reuse `FetchProviderKeys` instead? can we at least combine the 2 code blocks introduced in this PR into one?

Reply:
https://github.com/cyberark/conjur/pull/2143#discussion_r626502944
Will you be ok with it If I'll do the same as I mentioned in the comment here too?

Reply:
replied there